File: infrapulse_website/frontend/src/types/stats.ts
export interface CategoryStats {
  category: string;
  total: number;
  submitted: number;
  assigned: number;
  in_progress: number;
  resolved: number;
  avg_priority_score?: number | null;
}

export interface SystemStatsResponse {
  total_complaints: number;
  categories: CategoryStats[];
}

File: infrapulse_website/backend/app/schemas/stats.py
from typing import List, Optional
from pydantic import BaseModel

class CategoryStats(BaseModel):
    category: str
    total: int
    submitted: int
    assigned: int
    in_progress: int
    resolved: int
    avg_priority_score: Optional[float] = None

class SystemStatsResponse(BaseModel):
    total_complaints: int
    categories: List[CategoryStats]
